### ISSUE TITLE
Accessing registers values inside GPRSGX region

### ISSUE THREAD

**Issue by buttercutter** [utds3lab/sgx-lapd#1]
I am trying to print out the values of the registers inside GPRSGX region, but I could not both use the inline asm as you mentioned in your paper (%gs:0x20) and access the API which prints the SSA

1. In other words, why all the following inline asm code segments do not match the ESP value of the host (untrusted environment) ?

I have [attached my code as a zip file here](https://github.com/utds3lab/sgx-lapd/files/4412794/HelloEnclave.zip)

```
	// https://web.cse.ohio-state.edu/~lin.3021/file/RAID17.pdf#page=10
	// %gs:0x20 always points to theGPRSGX region

	// Layout of GPRSGX Portion of the State Save Area :
	// https://xem.github.io/minix86/manual/intel-x86-and-64-manual-vol3/o_fe12b1e2a880e0ce-1778.html

    uintptr_t offset = 176; // 32 bytes + 144 bytes = 0x20 + 0x90 = 0xB0
    long long ursp;

    //asm ("mov %%gs:%1, %0" : "=r"(ursp) : "m"(*(char *)offset)); // 2338810765860929536
    //asm ("mov %%gs:(%1), %0" : "=r"(ursp) : "r"(offset)); // -4290487885628440576
   //asm ("mov %%gs:0xB0, %0" : "=r"(ursp)); // 0
   asm ("mov %%gs:0x20 , %0" : "=r"(ursp)); // 140306321915720
```

2. Besides, why https://github.com/sangfansh/SGX101_sample_code/blob/master/HelloEnclave/Makefile#L139 does not link to https://github.com/intel/linux-sgx/search?q=ssa_gpr&unscoped_q=ssa_gpr ?

In other words, why is the Makefile I linked above does not link to trts library ?

```
    // read the outside stack address from current SSA
    thread_data_t *thread_data = get_thread_data();
    ssa_gpr_t *ssa_gpr = reinterpret_cast<ssa_gpr_t *>(thread_data->first_ssa_gpr);
    size_t addr = ssa_gpr->REG(sp_u);
```
